learning_resources_search/api.py

```diff
@@ -517,9 +545,118 @@ def execute_learn_search(search_params):
         )
         search = search.extra(aggs=aggregation_clauses)
 
+    return search
+
+
+def execute_learn_search(search_params):
+    """
+    Execute a learning resources search based on the query
+
+
+    Args:
+        search_params (dict): The opensearch query params returned from
+        LearningResourcesSearchRequestSerializer
+
+    Returns:
+        dict: The opensearch response dict
+    """
+
+    search = construct_search(search_params)
+
     return search.execute().to_dict()
 
 
+def subscribe_user_to_search_query(user, search_params):
+    from learning_resources_search.models import PercolateQuery
+
+    """
+    Subscribe a user to a search query
+
+
+    Args:
+        user: The User to subscribe
+        search_params (dict): The opensearch query params returned from
+        LearningResourcesSearchRequestSerializer
+
+    Returns:
+        dict: The opensearch response dict
+    """
+    adjusted_original_query = adjust_original_query_for_percolate(search_params)
+
+    percolate_query, _ = PercolateQuery.objects.get_or_create(
+        source_type=PercolateQuery.SEARCH_SUBSCRIPTION_TYPE,
+        original_query=adjusted_original_query,
+        query=adjust_query_for_percolator(adjusted_original_query),
+    )
+    if not percolate_query.users.filter(id=user.id).exists():
+        percolate_query.users.add(user)
+    return percolate_query
```

Comment:
You could add the call `percolate_query_saved_actions(percolate_query)` here now and get rid of the post_save signal.   Can be in a subsequent PR too.

Reply:
will leave this for a subsequent PR since currently there is an admin integration which I think will be handy in the next step.